src/pytest/__init__.py

```diff
@@ -46,8 +46,57 @@
 from _pytest.warning_types import PytestUnknownMarkWarning
 from _pytest.warning_types import PytestWarning
 
+# For mypy Any type checking purposes.
+# This file sets disallow_any_expr to ensure that the public API
+# does not have dynamic typing via Any. Manually using each public API
+# type as an expression to enforce this.
```

Comment:
Can this be moved to `__init__.pyi`?

Reply:
We can explicitly put types into an __init__.pyi, but that seems less desirable than having the code annotated inline. With the current setup, the code is annotated inline, and the 
 `disallow_any_expr` ensures that `Any` is not used as an annotation in this file.

`disallow_any_expr` was implemented into mypy here https://github.com/python/mypy/pull/3519
Notably, it requires that each item exists in the file as an expression. The import itself does not count as an expr here - so we need to have a series of `a = a` style expression assigments

Reply:
In that case I'd prefer to simply not use `disallow_any_expr` for now; it's a particularly un-pythonic change and we can easily revisit later.

Reply:
@nipunn1313 I've meant to move the "Manually using each public API type as an expression to enforce this." part of it.  It is only boilerplate, isn't it?

Reply:
`disallow_any_expr` is actually the most important piece - as it's the one that is guaranteeing that the public interface is typed.

The public interface is only the `__init__.py` file - so without this boilerplate - we're not actually adding much value.